### PR TITLE
feat(ui): Add Hindi localization language support (#7778).

### PR DESCRIPTION
**Pull request by adityathakurxd**
## Description

Added localization support for Hindi. Hindi is the main language of the Indian states of Uttar Pradesh, Haryana, Himachal Pradesh, and the capital Delhi in North India; Bihar and Jharkhand, and more.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/flutterfire_ui/lib/src/i10n/lang/hi.dart [hunk 1]
```diff
@@ -0,0 +1,263 @@
+import '../default_localizations.dart';
+
+class EnLocalizations extends FlutterFireUILocalizationLabels {
```
Comment (rahulraj64) [inline packages/flutterfire_ui/lib/src/i10n/lang/hi.dart:3]: Hello Aditya, shouldn't it be **HiLocalizations**?

File: packages/flutterfire_ui/lib/src/i10n/lang/hi.dart [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+  final String latitudeLabel;
+  @override
+  final String longitudeLabel;
+  @override
+  final String geopointLabel;
+  @override
+  final String referenceLabel;
+
+  const EnLocalizations({
+    this.emailInputLabel = 'Email',
```
Comment (rahulraj64) [inline packages/flutterfire_ui/lib/src/i10n/lang/hi.dart:172]: Seems there are a lot of texts yet to be translated to Hindi :) If this is a work in progress, please consider marking this as a draft :) 
(I'm not from Google. Just sharing some thoughts)

**Review by rahulraj64 — COMMENTED** [review]
Saw some texts yet pending translations.

File: packages/flutterfire_ui/lib/src/i10n/lang/hi.dart (hunk 1 above)
Comment (adityathakurxd) [reply packages/flutterfire_ui/lib/src/i10n/lang/hi.dart:3]: Yes Sorry about that I'll fix this.

File: packages/flutterfire_ui/lib/src/i10n/lang/hi.dart (hunk 2 above)
Comment (adityathakurxd) [reply packages/flutterfire_ui/lib/src/i10n/lang/hi.dart:172]: I had a question regarding that. I could see some other translations use Email as Email. I thought it'd do for Hindi as well. Should I like translate these too?

Comment (rahulraj64) [reply packages/flutterfire_ui/lib/src/i10n/lang/hi.dart:172]: I suggest adding native translations where ever possible. Also, I noticed that you have used English scripts all over the place for translation. Some languages like Bahasa, Spanish, etc use English script by default and this is not the case for Hindi. For example, 
**Phone number zaroori hai** - Wrong
**फ़ोन नंबर आवश्यक है** - Correct

Please note that I'm not a native Hindi speaker, these are some generic comments and I may not be able to review the translation itself

File: packages/flutterfire_ui/lib/src/i10n/lang/hi.dart [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    this.wrongOrNoPasswordErrorText =
+        'The password is invalid or the user does not have a password',
+    this.signInText = 'Sign in',
+    this.registerText = 'Register',
+    this.registerHintText = "Don't have an account?",
+    this.signInHintText = 'Already have an account?',
+    this.signOutButtonText = 'Sign out',
+    this.phoneInputLabel = 'Phone number',
+    this.phoneNumberInvalidErrorText = 'Phone number is invalid',
+    this.phoneNumberIsRequiredErrorText = 'Phone number zaroori hai',
```
Comment (rahulraj64) [inline packages/flutterfire_ui/lib/src/i10n/lang/hi.dart:210]: Use proper language script all over your translations. Currently, they are Hindi-In-English (Not proper Hindi translations)

File: packages/flutterfire_ui/lib/src/i10n/default_localizations.dart [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import 'package:flutterfire_ui/src/i10n/lang/hi.dart';
```
Comment (lesnitsky) [inline packages/flutterfire_ui/lib/src/i10n/default_localizations.dart:1]:
```suggestion
import 'lang/hi.dart';
```

File: packages/flutterfire_ui/lib/src/i10n/lang/hi.dart (hunk 1 above)
Comment (lesnitsky) [inline packages/flutterfire_ui/lib/src/i10n/lang/hi.dart:3]:
```suggestion
class HiLocalizations extends FlutterFireUILocalizationLabels {

```

**Review by lesnitsky — CHANGES_REQUESTED** [review]
Still several unresolved review comments.
@adityathakurxd would you be able to make relevant changes?

**Comment by adityathakurxd** [comment]
Yes, @lesnitsky I'll update with the required changes. Sorry for the delay.

File: packages/flutterfire_ui/lib/src/i10n/lang/hi.dart (hunk 2 above)
Comment (adityathakurxd) [reply packages/flutterfire_ui/lib/src/i10n/lang/hi.dart]: Hey @rahulraj64, I have made the changes. Could you please review it again? Thanks!

File: packages/flutterfire_ui/lib/src/i10n/lang/hi.dart [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+    this.passwordResetEmailSentText =
+        "हमने आपको आपका पासवर्ड रीसेट करने के लिए एक लिंक के साथ एक ईमेल भेजा है। कृपया अपनी ईमेल देखें।",
+    this.forgotPasswordHintText =
+        'अपना ईमेल प्रदान करें और हम आपको आपका पासवर्ड रीसेट करने के लिए एक लिंक भेजेंगे',
+    this.emailLinkSignInButtonLabel = 'मैजिक लिंक के साथ साइन इन करें',
+    this.signInWithEmailLinkViewTitleText = 'मैजिक लिंक के साथ साइन इन करें',
+    this.signInWithEmailLinkSentText =
+        "हमने आपको एक जादुई लिंक के साथ एक ईमेल भेजा है। अपना ईमेल जांचें और साइन इन करने के लिए लिंक का अनुसरण करें",
+    this.sendLinkButtonLabel = 'मैजिक लिंक भेजें',
+    this.arrayLabel = 'array',
```
Comment (lesnitsky) [inline packages/flutterfire_ui/lib/src/i10n/lang/hi.dart:243]: @adityathakurxd could these (lines 243 to 261) labels be translated as well?

Comment (adityathakurxd) [reply packages/flutterfire_ui/lib/src/i10n/lang/hi.dart]: I have updated the same.

**Comment by adityathakurxd** [comment]
Hi, @lesnitsky Could you tell me about the checks? How can I improve for them?

**Review by lesnitsky — APPROVED** [review]

**Comment by russellwheatley** [comment]
Hey @adityathakurxd, thanks for the PR! Do you mind resolving the merge conflict, please? 

Hey @rahulraj64, do you mind proofreading the update language translations in Hindi and approving if you find acceptable, please? Unfortunately, none of the maintainers I know speak Hindi. Thank you!

**Comment by adityathakurxd** [comment]
Hey @russellwheatley, I have resolved the merge conflicts.

**Comment by russellwheatley** [comment]
Hey @adityathakurxd, I don't think the merge conflict has been resolved.

**Comment by adityathakurxd** [comment]
Hey @russellwheatley, Sorry about that. I had last resolved the conflicts 8 days ago and #7709 was merged just after that which caused it again. Thank you! Let me know if there's anything else I need to do.

**Comment by russellwheatley** [comment]
Hey @adityathakurxd, do you mind getting a fellow developer who speaks Hindi to proofread & approve to ensure there are no mistakes (We all make them 😄 ), please?

**Comment by adityathakurxd** [comment]
Sure @russellwheatley I'll ask a few developers to review this.

**Comment by infiniteoverflow** [comment]
All the translations are perfect. 

@adityathakurxd just curious about the following : 

`verifyPhoneNumberButtonText` and `linkEmailButtonText` are both translated to **अगला**. Does this have to be like this in this context or can it be changed to **फ़ोन नंबर सत्यापित करें** and **लिंक ईमेल** ?

**Comment by adityathakurxd** [comment]
Hey, Aswin (@infiniteoverflow)
Thank you so much for the review! I had translated the original `en.dart` available [here](https://github.com/FirebaseExtended/flutterfire/blob/master/packages/flutterfire_ui/lib/src/i10n/lang/en.dart). Both the `verifyPhoneNumberButtonText` and `linkEmailButtonText` are button labels that say 'Next' in English so I thought it'd best to translate them to अगला just in case there are further steps to be performed.

**Review by russellwheatley — APPROVED** [review]
LGTM

**Pull request merged**